src/main/java/gregtech/integration/jei/basic/GTFluidVeinCategory.java

```diff
@@ -0,0 +1,158 @@
+package gregtech.integration.jei.basic;
+
+import gregtech.api.gui.GuiTextures;
+import gregtech.api.util.GTJEIUtility;
+import gregtech.api.util.GTStringUtils;
+import gregtech.api.util.GTUtility;
+import gregtech.api.worldgen.config.BedrockFluidDepositDefinition;
+import gregtech.api.worldgen.config.WorldGenRegistry;
+import mezz.jei.api.IGuiHelper;
+import mezz.jei.api.gui.IDrawable;
+import mezz.jei.api.gui.IGuiFluidStackGroup;
+import mezz.jei.api.gui.IRecipeLayout;
+import mezz.jei.api.ingredients.IIngredients;
+import mezz.jei.api.recipe.IRecipeWrapper;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.resources.I18n;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class GTFluidVeinCategory extends BasicRecipeCategory<GTFluidVeinInfo, GTFluidVeinInfo> {
+
+    protected final IDrawable slot;
+    private BedrockFluidDepositDefinition definition;
+    private String veinName;
+    private int weight;
+    private int[] yields; // the [minimum, maximum) yields
+    private int depletionAmount; // amount of fluid the vein gets drained by
+    private int depletionChance; // the chance [0, 100] that the vein will deplete by 1
+    private int depletedYield; // yield after the vein is depleted
+    private final int SLOT_CENTER = 79;
+    protected final Map<Integer, String> namedDimensions = WorldGenRegistry.getNamedDimensions();
+    private Supplier<List<Integer>> dimension;
+    private final int textStartX = 5;
+    private int weightLength;
+    private int minYieldLength;
+    private int maxYieldLength;
+    private int depletionChanceLength;
+    private int depletionAmountLength;
+    private int depletedYieldLength;
+    private final int startPosY = 40;
+
+
+    public GTFluidVeinCategory(IGuiHelper guiHelper) {
+        super("fluid_spawn_location",
+                "fluid.spawnlocation.name",
+                guiHelper.createBlankDrawable(176, 166),
```

Comment:
This category's size is excessively tall. It could likely be reduced to only ~110-120 pixels tall.

Reply:
The extra length is most likely fine because the dimension list can get quite large when a mod like Advanced Rocketry is installed, in which case most of the space will be used.